components/VM_Arm/src/main.c

```diff
@@ -620,6 +620,29 @@ static void irq_handler(void *data, ps_irq_acknowledge_fn_t acknowledge_fn, void
     }
 }
 
+/* Default RAM initialization. Modules can overwrite this weak function with a
+ * custom initialization. e.g. to use memory shared with other components or
+ * mapped on demand.
+ */
+WEAK int init_ram_module(vm_t *vm)
+{
+    if (0 == ram_size) {
+        ZF_LOGW("VM%d: no RAM defined", vm->vm_id);
+        return 0;
```

Comment:
I was under illusion that we agreed that all systems have some RAM, even if very small amount. Now this `0` seems to contradict that?

Reply:
I've kept this with the comment as sanity check. But I can remove it also completely.

Reply:
Another option is to put this check to `templates/seL4VMParameters.template.c`? Putting it in this function and having the option to overload the function would lead someone to think that non-zero RAM size is important only if the RAM is mapped in this way. But we already agree no matter how and where the RAM is mapped from, it needs to be non-zero in size?

And I have to nag about style issue once again. Could you use `vm_config->ram.size == 0` or `!vm_config->ram.size` instead of this, albeit technically superior but not traditional way to write the comparison? If you want the project to use this style, could you start a discussion on seL4 discourse for example and see whether there's support for this in the community? Up until then I think we should not deviate from the existing style.

Reply:
I will create a PR that adds a sanity check in `templates/seL4VMParameters.template.c` and don't print anything here.